### ISSUE TITLE
JS: "Uncaught TypeError: Cannot convert undefined or null to object" in pruneCIDs

### ISSUE THREAD

**Issue by henrik** [phoenixframework/phoenix_live_view#1052]
LV 0.14.0 from Hex. Have rebuilt assets.

See repro below. If you alternate clicking the "One" and "Two" links, you'll see this JS error on every click.

The `return` line here seems to be where the error happens:

``` js
                key: "pruneCIDs",
                value: function(e) {
                    var t = this;
                    e.forEach(function(e) {
                        return delete t.rendered.c[e]
                    })
```

The repro (note that `RemitWeb` is in a few places and likely needs find-and-replacing in your project – didn't see an easy way to avoid that): 

``` elixir
# Assumes this route:
live "/debug", DebugLive

defmodule MyItem do
  defstruct id: 0
end

defmodule RemitWeb.DebugLive do
  use Phoenix.LiveView

  @impl true
  def render(assigns) do
    ~L"""
    <%= live_render @socket, ItemsLive, id: "items_#{@number}" %>
    """
  end

  @impl true
  def mount(_params, _session, socket) do
    {:ok, socket}
  end

  @impl true
  def handle_params(params, _uri, socket) do
    {:noreply, assign(socket, number: params["number"])}
  end
end

defmodule ItemsLive do
  use Phoenix.LiveView
  alias RemitWeb.Router.Helpers, as: Routes

  @impl true
  def render(assigns) do
    ~L"""
    <p>
      <%= live_patch "One", to: Routes.live_path(@socket, RemitWeb.DebugLive, number: "one") %>
      <%= live_patch "Two", to: Routes.live_path(@socket, RemitWeb.DebugLive, number: "two") %>
    </p>

    <%= for item <- @items do %>
      <%= live_component @socket, ItemComponent, id: item.id %>
    <% end %>
    """
  end

  def mount(_params, session, socket) do
    {:ok, assign(socket,
      items: [%MyItem{id: 1}],
    )}
  end
end

defmodule ItemComponent do
  use Phoenix.LiveComponent

  @impl true
  def render(assigns) do
    ~L"""
    <p>hi</p>
    """
  end
end
```

**Comment by seb3s** [comment]
Hi Henrik,
looks very similar to the one I've submitted in #1048.
Was corrected by José. Can you check if 0.14.1 fixes it ?
Cheers,
Sebastien.

**Comment by chrismccord** [comment]
This should be fixed on 0.14.1, which was just released :) Please try and report back and I will reopen if needed. Thanks!

**Comment by henrik** [comment]
Yup, fixed in 0.14.1 :) Well caught, @seb3s!
